packages/docregistry/src/default.ts

```diff
@@ -573,8 +573,8 @@ class MimeDocumentFactory extends ABCWidgetFactory<MimeDocument, DocumentRegistr
   /**
    * Construct a new markdown widget factory.
    */
-  constructor(options: MimeDocumentFactory.IOptions) {
-    super({ ...options, readOnly: true });
+  constructor(options: MimeRendererFactory.IOptions) {
+    super(Private.createRegistryOptions(options));
```

Comment:
this indirection is needed because TS 2.4 is more strict about providing extra properties in object literals, and `MimeRendererFactory.IOptions` contains stuff which is not needed by the superclass options.